### ISSUE TITLE
IncompatibleClassChangeError failing with OpenJDK 14.0.1 OpenJ9, working with HotSpot

### ISSUE THREAD

**Issue by giacgbj** [jqno/equalsverifier#294]
Hi, using:

`EqualsVerifier.forClass(Foo::class.java).verify()	`
	
I'm experiencing the following issue:

```
EqualsVerifier found a problem in class Foo.
  -> class nl/jqno/equalsverifier/internal/lib/bytebuddy/matcher/FilterableList$AbstractBase has conflicting defaults for method forEach(Ljava/util/function/Consumer;)V
  Caused by: java.lang.IncompatibleClassChangeError: class nl/jqno/equalsverifier/internal/lib/bytebuddy/matcher/FilterableList$AbstractBase has conflicting defaults for method forEach(Ljava/util/function/Consumer;)V
  	at FooTest.testEqualsSymmetric(FooTest.kt:37)
```
  
using OpenJDK 14.0.1 **OpenJ9**, while it works using OpenJDK 14.0.1 **HotSpot**.

I'm using equalsverifier 3.1.13.

What could it be?

**Comment by jqno** [comment]
I'm not sure. What does the `Foo::class.java` syntax do again? I'm not
familiar with that.

Also, can you post a full `Foo` class that reproduces the issue? Something
that I can paste into IntelliJ? That would make it a lot easier for me to
figure out what's going on. Sadly, I don't have a lot of free time to spend
on EqualsVerifier these days.

**Comment by giacgbj** [comment]
The language is Kotlin.

The `::class.java` syntax is explained [here](https://kotlinlang.org/docs/reference/reflection.html#class-references)

The class under test is:
```
internal data class Foo(val content: ByteArray) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Foo

        if (!content.contentEquals(other.content)) return false

        return true
    }

    override fun hashCode(): Int {
        return content.contentHashCode()
    }
}
```

**Comment by jqno** [comment]
Ah, so you're doing Kotlin. That would have been a good thing to open with
😉. I've never used it though.

Is it allowed to override an internal Kotlin class? Because that's one of
the things equalsverifier does in the symmetry check.

Do you get this error on a Java class as well? Make sure you put it in the
same project so you get the same classpath.

Also, can you check if you have equalsverifier and/or bytebuddy on the
classpath twice?

**Comment by giacgbj** [comment]
I tried to convert, using IntelliJ IDEA, the Kotlin class to Java and I have the same issue.

However, I found out that:
- no error running them in IntelliJ IDEA
- error running it from Maven with other tests
- no error running it alone or with some other tests. It's not easy to find the minimum set of test because there are many

The error happens only using OpenJ9.

I don't have any dependency duplicated on the classpath.
